src/charm_state.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""This module defines the CharmState class which represent the state of the Flask charm."""
+
+import datetime
+import pathlib
+import typing
+
+
+class WebserverConfig(typing.NamedTuple):
+    """Represents the configuration values for a web server.
+
+    Attributes:
+        workers: The number of workers to use for the web server, or None if not specified.
+        threads: The number of threads per worker to use for the web server,
+            or None if not specified.
+        keepalive: The time to wait for requests on a Keep-Alive connection,
+            or None if not specified.
+        timeout: The request silence timeout for the web server, or None if not specified.
+    """
+
+    workers: int | None
+    threads: int | None
+    keepalive: datetime.timedelta | None
+    timeout: datetime.timedelta | None
+
+
+class CharmState:
+    """Represents the state of the Flask charm."""
+
+    def __init__(self, charm_config: typing.Mapping[str, str]):
+        """Initialize a new instance of the CharmState class.
+
+        Args:
+            charm_config: Charm configurations of the charm instance associated with this state.
+        """
+        self._charm_config = charm_config
```

Comment:
We should not store all the charm config in the State, we only take the elements we need

Reply:
As I mentioned before, the process of determining which configuration options make up the charm state should be an internal implementation detail of CharmState, separated from the concerns of FlaskCharm.

Reply:
See the other discussion, I close this comment to avoid duplicates